src/ILLink.RoslynAnalyzer/DataFlow/LocalDataFlowVisitor.cs

```diff
@@ -0,0 +1,185 @@
+	// - field
+	// - parameter
+	// - method return
+	public abstract class LocalDataFlowVisitor<TValue, TValueLattice> : OperationVisitor<LocalState<TValue>, TValue>,
+		ITransfer<BlockProxy, LocalState<TValue>, LocalStateLattice<TValue, TValueLattice>>
+		where TValue : IEquatable<TValue>
+		where TValueLattice : ILattice<TValue>
+	{
+		protected readonly LocalStateLattice<TValue, TValueLattice> LocalStateLattice;
+
+		protected readonly OperationBlockAnalysisContext Context;
+
+		protected TValue TopValue => LocalStateLattice.Lattice.ValueLattice.Top;
+
+		public LocalDataFlowVisitor (LocalStateLattice<TValue, TValueLattice> lattice, OperationBlockAnalysisContext context) =>
+			(LocalStateLattice, Context) = (lattice, context);
+
+		public void Transfer (BlockProxy block, LocalState<TValue> state)
+		{
+			foreach (IOperation operation in block.Block.Operations)
+				Visit (operation, state);
+
+			// Blocks may end with a BranchValue computation. Visit the BranchValue operation after all others.
+			IOperation? branchValueOperation = block.Block.BranchValue;
+			if (branchValueOperation != null) {
+				var branchValue = Visit (branchValueOperation, state);
+
+				// BranchValue may represent a value used in a conditional branch to the ConditionalSuccessor - if so, we are done.
+
+				// If not, the BranchValue represents a return or throw value associated with the FallThroughSuccessor of this block.
+				// (ConditionalSuccessor == null iff ConditionKind == None).
+				if (block.Block.ConditionKind == ControlFlowConditionKind.None) {
+					// This means it's a return value or throw value associated with the fall-through successor.
+
+					// Return statements with return values are represented in the control flow graph as
+					// a branch value operation that computes the return value. The return operation itself is the parent
+					// of the branch value operation.
+
+					// Get the actual "return Foo;" operation (not the branch value operation "Foo").
+					// This should be used as the location of the warning. This is important to provide the right
+					// warning location and because warnings are disambiguated based on the operation.
+					var parentSyntax = branchValueOperation.Syntax.Parent;
+					if (parentSyntax == null)
+						throw new InvalidOperationException ();
+
+					var parentOperation = Context.Compilation.GetSemanticModel (branchValueOperation.Syntax.SyntaxTree).GetOperation (parentSyntax);
+
+					if (parentOperation is IThrowOperation)
+						throw new NotImplementedException ();
+
+					if (parentOperation is not IReturnOperation returnOperation)
+						throw new InvalidOperationException ();
+
+					HandleReturnValue (branchValue, returnOperation);
+				}
+			}
+		}
+
+		public abstract void HandleAssignment (TValue source, TValue target, IOperation operation);
+
+		public abstract void HandleReceiverArgument (TValue receiver, IInvocationOperation operation);
```

Comment:
I assume this is some Roslyn term, but I think it might be good to have a comment about what this means.
Reading the code, this is basically a call to a method, right?

Reply:
Yup, specifically an instance method call. I added a comment.